design/baremetal-operator/firmware-interface.md

```diff
@@ -0,0 +1,222 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Support Firmware Interface
+
+## Status
+
+implementable
+
+## Summary
+
+This design document describes a Metal3 API for using the Firmware Interface
+feature added to Ironic, this will allow updates of firmware of bmc and bios
+in BMH that are compatible with Redfish.
+
+## Motivation
+
+When provisioning new machines to clusters, it may be necessary to upgrade
+or downgrade the servers BIOS and firmware to specific versions to ensure
+the server is configured as it is in their validated pattern.
+
+Redfish has a common API that allow users to execute a firmware update, the
+[proposal](https://review.opendev.org/c/openstack/ironic-specs/+/878505) was
+accepted in the Ironic community, and is already implemented.
+
+### Goals
+
+This proposal does not attempt to fulfill the following goals:
+
+- Allow Firmware updates for the BMC and BIOS in hardware that compatible
+  with Redfish.
+
+### Non-Goals
+
+- Update Firmware using non Redfish means
```

Comment:
we should probably keep the word "firmware" not capitalized, except when at the start of a sentence :)

Reply:
done